include/tracing_impl/nvtx.h

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2022-2024 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
+
+#ifndef NVTX_H
+#define NVTX_H
+
+#if HAVE_NVTX_TRACING
+#include "nvToolsExt.h"
+
+static inline void nvtx_mark_domain(nvtxDomainHandle_t domain, const char* name, uint32_t color)
+{
+	const nvtxEventAttributes_t eventAttrib = {
+		.version = NVTX_VERSION,
+		.size = NVTX_EVENT_ATTRIB_STRUCT_SIZE,
+		.colorType = NVTX_COLOR_ARGB,
+		.color = color,
+		.messageType = NVTX_MESSAGE_TYPE_ASCII,
+		.message = { .ascii = name },
+	};
```

Comment:
Not for this PR, but we should make use of the payload field in the event attribute to get some request-specific information passed up to the tool...

Reply:
Yeah, agree that would be useful. Future PR.